### PR TITLE
Add tips for trouble shooting

### PR DESCRIPTION
**Pull request by jichenjc**
time to time we have some questions on how to debug issue, add this for some debug tip

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -197,3 +197,26 @@ your openstack Cluster API Kubernetes cluster.
...
+
+### Trouble shooting
+
+This guide (based on minikube and others should be similar) explains general info on how to debug issues if cluster failed to create.
+
+1. Get openstack contain name, the output depends on the system you are running.
+   the `minikube.kubeconfig` which is bootstrap cluster's kubeconfig by default locates at `cmd/clusterctl` folder.
+
+   ```
+   # kubectl --kubeconfig minikube.kubeconfig get pods --all-namespaces
```
Comment (gyliu513) [inline README.md]: How about 

```
kubectl get pods -n openstack-provider-system
```

File: README.md [hunk 2]
```diff
@@ -197,3 +197,26 @@ your openstack Cluster API Kubernetes cluster.
...
+   system                      controller-manager-0                     1/1     Running   0          27m
+   ```
+
+2. Get log of clusterapi-controllers-xxxxxxxx-xxxxx
+
+   ```
+   # kubectl --kubeconfig minikube.kubeconfig log clusterapi-controllers-xxxxxxxxx-xxxxx -n openstack-provider-system
+   ```
+
+3. If above log is not enough, you might want check openstack log as well, e.g. `/var/log/nova/nova-api.log` and `/var/log/nova/nova-compute.log` (by default) if create machine has some issue.
```
Comment (gyliu513) [inline README.md]: Sometimes, the end user may not have permission to check those logs of nova. Can we remove this and only keep the controller log?

File: Gopkg.lock [hunk 3]
```diff
@@ -178,7 +178,7 @@
 
 [[projects]]
   branch = "master"
-  digest = "1:e6e8950a9abc02d8cef26daada00a1ac0f0ee8e8398bfb84ed105e3e6e1bc87b"
+  digest = "1:ffc6d1eb832bc164c2292015fe711975ef6d312887c84cdaff9a8f23b4325e3f"
```
Comment (gyliu513) [inline Gopkg.lock]: Why change this?

Comment (jichenjc) [reply Gopkg.lock]: done

**Comment by gyliu513** [comment]
@jichenjc I found the PR was not updated, am I missing anything? ;-)

**Comment by jichenjc** [comment]
@gyliu513 I might made a mistake on the branch... will submit again

**Comment by hidekazuna** [comment]
@jichenjc Since we will add more contents into this, it may be better to write trouble_shooting.md under docs directory separately. If do so, add the link under Getting started section name like:

> This guide explains how to create and delete kubernetes cluster using this provider.
> If unexpected error occurred, see `trouble_shooting.md`.

**Comment by jichenjc** [comment]
works for me ,thanks for the advice

File: docs/trouble_shooting.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
+### Trouble shooting
+
```
Comment (gyliu513) [inline docs/trouble_shooting.md]: ## Trouble shooting

### get log of clusterapi-controllers containers

File: docs/trouble_shooting.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+# Trouble shooting
+
+## get log of clusterapi-controllers containers
```
Comment (gyliu513) [inline docs/trouble_shooting.md]: s/get log of clusterapi-controllers containers/Get log of clusterapi-controllers containers

File: docs/trouble_shooting.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+This guide (based on minikube and others should be similar) explains general info on how to debug issues if cluster failed to create.
```
Comment (gyliu513) [inline docs/trouble_shooting.md]: How about move this between `Trouble shooting` and `Get log of clusterapi-controllers containers`?

File: README.md [hunk 7]
```diff
@@ -197,3 +197,7 @@ your openstack Cluster API Kubernetes cluster.
    ```bash
    rm -rf $HOME/.ssh/openstack_tmp*
    ```
+
+### Trouble shooting
+
+please refer to [Trouble shooting documentation](docs/config.md) for further info.
```
Comment (hidekazuna) [inline README.md]: config.md should be trouble_shooting.md.

File: docs/trouble_shooting.md [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+1. Get openstack contain name, the output depends on the system you are running.
```
Comment (hidekazuna) [inline docs/trouble_shooting.md]: What is 'openstack contain name'? container?

File: README.md [hunk 9]
```diff
@@ -197,3 +197,7 @@ your openstack Cluster API Kubernetes cluster.
    ```bash
    rm -rf $HOME/.ssh/openstack_tmp*
    ```
+
+### Trouble shooting
+
+please refer to [Trouble shooting documentation](docs/trouble_shooting.md) for further info.
```
Comment (hidekazuna) [inline README.md]: s/please/Please/

**Pull request merged**